src/app/campaign/store/actions/campaign-action.service.ts

```diff
@@ -3,17 +3,20 @@ import { Store, select } from '@ngrx/store';
 import { Subscription } from 'rxjs';
 import { first, filter } from 'rxjs/operators';
 import * as allocationPreviewActions from './allocation-preview-action.creator';
+import * as availabilityActions from './availability-action.creator';
 import * as campaignActions from './campaign-action.creator';
 import { Flight } from '../models';
 import { selectCampaignId, selectCampaignWithFlightsForSave, selectRoutedFlight, selectRoutedLocalFlight } from '../selectors';
-import { CampaignStoreService } from '../../../core';
 
 @Injectable()
 export class CampaignActionService implements OnDestroy {
   private flightSub: Subscription;
+  // current flight id only to be used for checking against changed flight id to load availability on flight tab/route change
+  // for any other current/routed flight id needs, please pipe from the selectRoutedFlightId selector
+  // one reason the routed flight id selector is not used here is that is would fire on navigation before the flight is actually loaded
```

Comment:
felt like I had to justify and provide some caution on this state variable, seems like a bit of an anti pattern

Reply:
I think I could set this up with some of the selectors just not something I explored too much yet